src/modules/submissions.js

```diff
@@ -697,6 +697,39 @@
 		return this.text;
 	};
 
+	AFCH.Text.prototype.updateShortDescription = function ( existingShortDescription, newShortDescription ) {
+		var shortDescTemplateExists = /\{\{[Ss]hort ?desc(ription)?\s*\|/.test( this.text );
+		var shortDescExists = !!existingShortDescription;
+
+		if ( newShortDescription ) {
+			// 1. No shortdesc - insert the one provided by user
+			if ( !shortDescExists ) {
+				this.prepend( '{{Short description|' + newShortDescription + '}}\n' );
+
+			// 2. Shortdesc exists from {{short description}} template - replace it
+			} else if ( shortDescExists && shortDescTemplateExists ) {
+				this.text = this.text.replace( /\{\{[Ss]hort ?desc(ription)?\s*\|.*?\}\}\n*/g, '' );
+				this.prepend( '{{Short description|' + newShortDescription + '}}\n' );
+
+			// 3. Shortdesc exists, but not generated by {{short description}}, if the user
+			//  has changed the value, save the new value
+			} else if ( shortDescExists && existingShortDescription !== newShortDescription ) {
+				this.prepend( '{{Short description|' + newShortDescription + '}}\n' );
+				// FIXME: but this value won't actually be recognized because the template that generates
```

Comment:
Could you do a regex search for the existing template and remove it, or would that not work?